modules/querier/querier.go

```diff
@@ -573,12 +584,23 @@ func (q *Querier) SearchTagsV2(ctx context.Context, req *tempopb.SearchTagsReque
 	limit := q.limits.MaxBytesPerTagValuesQuery(userID)
 	distinctValues := collector.NewScopedDistinctString(limit)
 
+	stopEarly := false
 	for _, resp := range lookupResults {
 		for _, res := range resp.response.(*tempopb.SearchTagsV2Response).Scopes {
 			for _, tag := range res.Tags {
 				distinctValues.Collect(res.Name, tag)
+				if distinctValues.Exceeded() {
+					stopEarly = true
```

Comment:
i don't mind this, but this is also a case where the dreaded goto statement can make sense. wdyt?

Reply:
yeah, goto is clean and less branching as well.